src/ray/raylet/node_manager.cc

```diff
@@ -1491,16 +1499,22 @@ bool NodeManager::AssignTask(const Task &task) {
   auto message = protocol::CreateGetTaskReply(fbb, spec.ToFlatbuffer(fbb),
                                               fbb.CreateVector(resource_id_set_flatbuf));
   fbb.Finish(message);
-  // Give the callback a copy of the task so it can modify it.
-  Task assigned_task(task);
   worker->Connection()->WriteMessageAsync(
       static_cast<int64_t>(protocol::MessageType::ExecuteTask), fbb.GetSize(),
-      fbb.GetBufferPointer(), [this, worker, assigned_task](ray::Status status) mutable {
+      fbb.GetBufferPointer(), [this, worker, task](ray::Status status) {
         if (status.ok()) {
-          auto spec = assigned_task.GetTaskSpecification();
+          auto spec = task.GetTaskSpecification();
           // We successfully assigned the task to the worker.
           worker->AssignTaskId(spec.TaskId());
           worker->AssignDriverId(spec.DriverId());
+
+          // Make a copy of the task to add to the lineage cache and eventually
+          // flush to the GCS.
+          // TODO(swang): We use a copy of the task so that for actor tasks, we
+          // can keep the original execution dependencies in the copy in the
+          // scheduling queue, but ideally the task in the lineage cache would
+          // match the queued task exactly.
```

Comment:
Yeah this is a little unfortunate...we need access to the original execution dependency because that is the object that gets released in the actor's frontier once the task completes. I thought about doing the dummy object accounting when the task is assigned, but you don't know yet whether the task succeeded.

Reply:
Oh hmm actually just realized that we can separate the logic so that we release the previous object when the task is assigned and add the new object when the task finishes. Then I can get rid of this code. :)